### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by AmyNguyen1130**
@PhuongNguyen @thuyct99. Please consider my exercise2 and comment if I have mistakes in code. thank you very much.

Exercise 2
"Description"
Write a function that takes in a string of one or more words, and returns the same string, but with all five or more letter words reversed (Just like the name of this Kata). Strings passed in will consist of only letters and spaces. Spaces will be included only when more than one word is present.

Examples: 
spinWords( "Hey fellow warriors" ) => returns "Hey wollef sroirraw" 
spinWords( "This is a test") => returns "This is a test" 
spinWords( "This is another test" )=> returns "This is rehtona test"

### CODE REVIEW

File: src/codeenginestudio/huong/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+public class Exercise2 {
+    public static void main(String[] args) {
+        Exercise2 exercise2 = new Exercise2();
+        String input = "Hey fellow warriors";
+        System.out.println("input  : " + input);
+        System.out.println("output : " + exercise2.reversedSomeWords(input));
+    }
+
+    private boolean checkLength(String input) {
+        if (input.length() >= 5)
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: Please make "5" to be a constant value.

File: src/codeenginestudio/huong/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        return joinArrayToString(strInput);
+    }
+
+    private void reverseWord(String[] strArray, int index) {
+        char[] arrWord = strArray[index].toCharArray();
+        char[] newArrWord = new char[arrWord.length];
+
+        for (int i = 0; i <= arrWord.length - 1; i++) {
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java:39]: Another way, you can use "reverse()" existing method.

File: src/codeenginestudio/huong/Exercise2.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            newArrWord[arrWord.length - i - 1] = arrWord[i];
+        }
+
+        strArray[index] = String.valueOf(newArrWord);
+    }
+
+    private String[] spiltStringToArray(String input) {
+        String[] strArray = input.split("\\s");
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: You can return immedially in here `return input.split(""\\s);`. Same for another methods

Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: Same above. Return immediately.

File: src/codeenginestudio/huong/Exercise2.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        Exercise2 exercise2 = new Exercise2();
+        String input = "Hey fellow warriors";
+        System.out.println("input  : " + input);
+        System.out.println("output : " + exercise2.reversedSomeWords(input));
+    }
+
+    private static final Integer BOUNDARY_MAX = 5;
+
+    private boolean checkLength(String input, int boundaryMax) {
+        if (input.length() >= boundaryMax)
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: Please add `{` after `if`

File: src/codeenginestudio/huong/Exercise2.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        for (int i = 0; i <= arrWord.length - 1; i++) {
+            newArrWord[arrWord.length - i - 1] = arrWord[i];
+        }
+
+        strArray[index] = String.valueOf(newArrWord);
+    }
+
+    private String[] spiltStringToArray(String input) {
+        return input.split("\\s");
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: Please make `'\\s` to constant variable

File: src/codeenginestudio/huong/Exercise2.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    private static final Integer BOUNDARY_MAX = 5;
+
+    private boolean checkLength(String input, int boundaryMax) {
+        if (input.length() >= boundaryMax)
+            return true;
+        return false;
+    }
+
+    public String joinArrayToString(String[] strArray) {
+        return String.join(" ", strArray);
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise2.java]: Please make `" "` to constant variable

**Comment by haubuicodeengine** [comment]
Well done.

**Comment by haubuicodeengine** [comment]
It is merged. Thank you.

**Comment by AmyNguyen1130** [comment]
Thank you very much @haubuicodeengine

**Pull request merged**